ql/src/java/org/apache/hadoop/hive/ql/parse/MergeSemanticAnalyzer.java

```diff
@@ -297,15 +309,16 @@ private boolean handleCardinalityViolation(StringBuilder rewrittenQueryStr, ASTN
     //this is a tmp table and thus Session scoped and acid requires SQL statement to be serial in a
     // given session, i.e. the name can be fixed across all invocations
     String tableName = "merge_tmp_table";
+    List<String> sortKeys = columnAppender.getSortKeys();
```

Comment:
Please talk with @lcspinter about the sorted Iceberg tables. How the 2 PR will work together.

Thanks,
Peter

Reply:
We discussed with @lcspinter that his patch does independent changes.
I also applied both his and this patch to the same branch locally and wrote a test case for updating sorted table. It works only for non vectorized mode.
The reason why it does not work in vectorized mode is not related to this patch. It depends on the setting of `iceberg.mr.in.memory.data.model`
It is set to `HIVE` when vectorization is enabled:
https://github.com/apache/hive/blob/855c0642342f83a80f30ecf41f1f7f08048d6f80/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java#L158
Delete delta files are not read with this setting.